fastify.d.ts

```diff
@@ -261,7 +261,10 @@ declare namespace fastify {
      * Registers a listener function that is invoked when all the plugins have
      * been loaded. It receives an error parameter if something went wrong.
      */
-    ready(readyListener?: () => void): void
+    ready(): Promise<FastifyInstance<HttpServer, HttpRequest, HttpResponse>>
+    ready(readyListener?: (err: Error) => void): void
+    ready(readyListener?: (err: Error, done: Function) => void): void
+    ready(readyListener?: (err: Error, context: FastifyInstance<HttpServer, HttpRequest, HttpResponse>, done: Function) => void): void
```

Comment:
`readyListener` is not optional for the callback cases so the `?` shouldn't be needed.

Reply:
You're right, good catch!